components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -11,31 +11,52 @@ import kotlinx.coroutines.ObsoleteCoroutinesApi
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.newSingleThreadContext
 import kotlinx.coroutines.runBlocking
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.ConcurrentLinkedQueue
 
 @ObsoleteCoroutinesApi
 internal object Dispatchers {
     class WaitableCoroutineScope(val coroutineScope: CoroutineScope) {
         // When true, jobs will be run synchronously
         internal var testingMode = false
 
+        // When true, jobs will be queued and not ran until triggered by calling
+        // flushQueuedInitialTasks()
+        @Volatile
+        internal var queueInitialTasks = true
+
+        // Use a [ConcurrentLinkedQueue] to take advantage of it's thread safety and no locking
+        internal val taskQueue: ConcurrentLinkedQueue<() -> Unit> = ConcurrentLinkedQueue()
+
+        private val logger = Logger("glean/Dispatchers")
+
+        companion object {
+            // This value was chosen in order to allow several tasks to be queued for execution but
+            // still be conservative of memory.  This queue size is important for cases where
```

Comment:
nit: double whitespace between "...memory." and "This queue..."

Reply:
I'll go ahead and remove the extra space, but I will also leave you with this article ;)   

https://www.theatlantic.com/science/archive/2018/05/two-spaces-after-a-period/559304/